test/array-data-provider.test.js

```diff
@@ -224,45 +224,28 @@ describe('invalid paths', () => {
   describe('invalid filters paths', () => {
     let filter;
 
-    beforeEach((done) => {
+    beforeEach(() => {
       filter = grid.querySelector('vaadin-grid-filter');
-      listenOnce(grid, 'filter-changed', () => {
-        done();
-      });
       filter.path = '';
+      filter._debouncerFilterChanged.flush();
     });
 
-    it('should warn about invalid path with undefined parent property', (done) => {
-      listenOnce(filter, 'filter-changed', () => {
-        setTimeout(() => {
-          expect(console.warn.called).to.be.true;
-          done();
-        });
-      });
-
+    it('should warn about invalid path with undefined parent property', () => {
       filter.path = 'foo.bar';
+      filter._debouncerFilterChanged.flush();
+      expect(console.warn.called).to.be.true;
     });
 
-    it('should not warn about undefined values with defined parent property', (done) => {
-      listenOnce(filter, 'filter-changed', () => {
-        setTimeout(() => {
-          expect(console.warn.called).to.be.false;
-          done();
-        });
-      });
-
+    it('should not warn about undefined values with defined parent property', () => {
       filter.path = 'name.foo';
+      filter._debouncerFilterChanged.flush();
```

Comment:
nit: is it intentional to flush a specific debouncer instead of running `flushGrid(grid)`?

Reply:
Yes, this debouncer is not present in `flushGrid` method. 
Alternatively we could use fake timers, but that would mean having to use magic numbers in tests.